### PR TITLE
feat: improve slow queries options deserialization

### PR DESCRIPTION
**Pull request by killme2008**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

#6697 

## What's changed and what's your intention?

This PR improves slow query options deserialization by adding better handling for empty strings and missing fields in configuration options, making them more robust when deserializing from JSON.

* Adds serde attributes to handle empty strings as default values for enum fields
* Introduces default functions for `SlowQueryOptions` fields to improve maintainability
* Adds comprehensive test coverage for deserialization edge cases

## PR Checklist
Please convert it to a draft if some of the following conditions are not met.

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.
- [ ] API changes are backward compatible.
- [ ] Schema or data changes are backward compatible.

### CODE REVIEW

File: src/common/telemetry/src/logging.rs [hunk 1]
```diff
@@ -96,49 +98,70 @@ pub enum OtlpExportProtocol {
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct SlowQueryOptions {
```
Comment (evenyag) [inline src/common/telemetry/src/logging.rs]: Why not derive `serde(default)` for the struct so we don't have to set default for each field?
```suggestion
#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
#[serde(default)]
pub struct SlowQueryOptions {
```

Comment (killme2008) [reply src/common/telemetry/src/logging.rs]: Good catch! I forgot this feature.

Comment (killme2008) [reply src/common/telemetry/src/logging.rs]: Addressed https://github.com/GreptimeTeam/greptimedb/pull/6734/commits/9890173841f2a6d988f37d254a93e54cafb61bc8

**Review by evenyag — APPROVED** [review]
